### PR TITLE
NODE-2516 Improve invoke fail/reject tests

### PR DESCRIPTION
**Pull request by ivan-mashonskiy**

### CODE REVIEW

File: node/src/test/scala/com/wavesplatform/features/RideV6FailRejectTest.scala [hunk 1]
```diff
@@ -175,96 +286,116 @@ class RideV6ActivationTest extends FreeSpec with WithDomain with OptionValues wi
...
-               | strict to = Address(base58'$bobAddr')
-               | [${(1 to 101).map(_ => actionSrc).mkString(", ")}]
-               | """.stripMargin
+      } ++
+        dataEntries.map { case (entryType, entry) =>
+          Case(
+            s"NODE-546 If an invoke writes an empty $entryType key to the state (invoke version >= 2)",
+            "Empty keys aren't allowed in tx version >= 2",
+            mkDAppFunc(entry),
+            ethInvokeTx = None
```
Comment (vsuharnikov) [inline node/src/test/scala/com/wavesplatform/features/RideV6FailRejectTest.scala:321]: ?

Comment (vsuharnikov) [inline node/src/test/scala/com/wavesplatform/features/RideV6FailRejectTest.scala:321]: A case for `EthereumTransaction` is missing.

File: node/src/test/scala/com/wavesplatform/features/RideV6FailRejectTest.scala [hunk 2]
```diff
@@ -175,96 +286,116 @@ class RideV6ActivationTest extends FreeSpec with WithDomain with OptionValues wi
...
-            .signWith(alice.privateKey)
+          invokeTx = TxHelpers
+            .invoke(
+              dApp = aliceAddr,
+              invoker = alice,
+              func = Some("foo"),
+              fee = 3.waves,
+              payments = Seq(InvokeScriptTransaction.Payment(1, Waves))
+            ),
+          ethInvokeTx = None
```
Comment (vsuharnikov) [inline node/src/test/scala/com/wavesplatform/features/RideV6FailRejectTest.scala:398]: A case for `EthereumTransaction` is missing.

File: node/src/test/scala/com/wavesplatform/state/diffs/ci/RideV5FailRejectTest.scala [hunk 3]
```diff
@@ -124,6 +123,9 @@ class InvokeFailAndRejectTest extends PropSpec with WithDomain {
       d.appendBlock(setScript(secondSigner, dApp))
       d.appendAndAssertFailed(invokeTx, "Transaction is not allowed by script of the asset")
     }
+
+    // TODO: move test after bug fix
+    //  after RideV6 failing of negative balance check before execution of ScriptTransfer asset script always leads to invoke rejection
```
Comment (vsuharnikov) [inline node/src/test/scala/com/wavesplatform/state/diffs/ci/RideV5FailRejectTest.scala]: Do we have a task for this? Could you add a task number here, e.g. `NODE-XXXX`?

**Review by vsuharnikov — CHANGES_REQUESTED** [review]

File: node/src/test/scala/com/wavesplatform/features/RideV6FailRejectTest.scala (hunk 1 above)
Comment (vsuharnikov) [reply node/src/test/scala/com/wavesplatform/features/RideV6FailRejectTest.scala:321]: Well, EthereumTransaction doesn't have the second version, closing

File: node/src/test/scala/com/wavesplatform/features/RideV6FailRejectTest.scala (hunk 2 above)
Comment (vsuharnikov) [reply node/src/test/scala/com/wavesplatform/features/RideV6FailRejectTest.scala:398]: Well, it is too hard to do (impossible?) :) Closing

**Review by vsuharnikov — APPROVED** [review]

**Review by yvaskov — APPROVED** [review]

**Review by phearnot — APPROVED** [review]

**Pull request merged**